app/com/gu/floodgate/reindex/ProgressTracker.scala

```diff
@@ -106,9 +107,14 @@ class ProgressTracker(ws: WSClient, runningJobService: RunningJobService, jobHis
 
       case InProgress =>
         val runningJobUpdate = RunningJob(runningJob.contentSourceId, runningJob.contentSourceEnvironment, progress.documentsIndexed, progress.documentsExpected, runningJob.startTime, runningJob.rangeFrom, runningJob.rangeTo)
-        runningJobService.updateRunningJob(runningJob.contentSourceId, runningJob.contentSourceEnvironment, runningJobUpdate)
-        scheduleNextUpdate(contentSource, runningJob)
+        val periodIndexing = Minutes.minutesBetween(runningJobUpdate.startTime, DateTime.now())
 
+        if (runningJobUpdate.documentsIndexed == 0 && (periodIndexing.getMinutes - 10) >= 0) {
```

Comment:
You could assign the predicate here to a clearly named val for readability here.

Reply:
Where is `10` coming from?

Reply:
I wonder if `runningJobUpdate` should be defined inside the `if` clause?

Reply:
Just an arbitrary choice - I looked at history of previous reindexes and ran a few in `CODE` and from that it looks as if all would have triggered reindex of one document by then.